util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/WindingOrder.java

```diff
@@ -0,0 +1,205 @@
+			boolean counterClockWise) {
+
+		// Checking and reversing geometry
+		if (isCounterClockwise(linearRing) == counterClockWise)
+			return linearRing;
+		else
+			return (LinearRing) linearRing.reverse();
+	}
+
+	/**
+	 * Unify order of the polygon as counterClockwise or not including all its
+	 * holes.
+	 * 
+	 * @param poly Polygon object for unifying
+	 * @param counterClockWise boolean value. true, if want shell of Polygon as
+	 *            counter clock wise and holes as clockwise, else false.
+	 * @return Polygon unified object.
+	 */
+	public static Polygon unifyWindingOrderForPolyGon(Polygon poly, boolean counterClockWise) {
+
+		// Checking and reversing Shell
+		LinearRing shell = unifyWindingOrderForLinearRing((LinearRing) poly.getExteriorRing(),
+				counterClockWise);
+
+		// Checking and reversing Holes
+		LinearRing holes[] = new LinearRing[poly.getNumInteriorRing()];
+		for (int i = 0; i < poly.getNumInteriorRing(); i++) {
+			holes[i] = unifyWindingOrderForLinearRing((LinearRing) poly.getInteriorRingN(i),
+					!counterClockWise);
+		}
+
+		// Create New Polygon using unified shell and holes
+		Polygon tempPoly = factory.createPolygon(shell,
+				poly.getNumInteriorRing() > 0 ? holes : null);
+
+		return tempPoly;
+	}
+
+	/**
+	 * Unify order of the Multipolygon including all polygons and holes in it.
+	 * 
+	 * @param multiPoly Multipolygon object for unifying it.
+	 * @param counterClockWise boolean value. true, if want shell of
+	 *            multipolygon as counter clock wise and holes as clockwise,
+	 *            else false.
+	 * @return Multipolygon unified Object
+	 */
+	public static MultiPolygon unifyWindingOrderForMultiPolygon(MultiPolygon multiPoly,
+			boolean counterClockWise) {
+
+		// get no of polygons in MultiPolygon
+		int noOfPolygons = multiPoly.getNumGeometries();
+
+		Polygon[] revGeoms = new Polygon[noOfPolygons];
+		// Unify each inner polygon one by one
+		for (int i = 0; i < noOfPolygons; i++) {
+			revGeoms[i] = unifyWindingOrderForPolyGon((Polygon) multiPoly.getGeometryN(i),
+					counterClockWise);
+		}
+		return factory.createMultiPolygon(revGeoms);
+	}
+
+	/**
+	 * Unify order of the GeometryCollection including all Geometries in it.
+	 * 
+	 * @param geoCollection GeometryCollection object for unifying.
+	 * @param counterClockWise boolean value. true, if want all geometry object
+	 *            * as counter clock wise, else false.
+	 * @return GeometryCollection unified Object
+	 */
+	public static GeometryCollection unifyWindingOrderForGeometryCollection(
+			GeometryCollection geoCollection, boolean counterClockWise) {
+
+		// get no of geometries in GeometryCollection
+		int noOfGeoms = geoCollection.getNumGeometries();
+
+		Geometry[] revGeoms = new Geometry[noOfGeoms];
+		// Unify each geometry one by one
+		for (int i = 0; i < noOfGeoms; i++) {
+			Geometry geometry = geoCollection.getGeometryN(i);
+			if (geometry instanceof MultiPolygon) {
```

Comment:
Instead of having the conditions for the different geometry types duplicated here, you could just call `unifyWindingOrder(geometry, counterClockWise)` for the individual geometries in the collection.